### PR TITLE
Feature(Issue #8): As a user, I want to mark an item on my shopping list as purchased so the app can learn how often I buy different items.

### LINKED ISSUE #8: 8. As a user, I want to mark an item on my shopping list as purchased so the app can learn how often I buy different items.
AC:

* User is able to tap a checkbox or similar UI element to mark an item in the list as purchased
* Item should be shown as checked for 24 hours after the purchase is made (i.e. we assume the user does not need to buy the item again for at least 1 day)

### PR DESCRIPTION
**Pull request by daredtech**
## Description

Added a new component <Item/> to store and display a single item from the database. Clicking on the checkbox near the item updates the `lastPurchaseDate` field with a current timestamp. When displaying the items in the list, if the `lastPurchaseDate` was less than 24 hours ago, the item would appear checked and crossed. The assumption is that the user does not need to buy the item again for at least one day. 

## Related Issue

Closes https://github.com/the-collab-lab/tcl-11-smart-shopping-list/issues/8

## Acceptance Criteria

- [x] User is able to tap a checkbox or similar UI element to mark an item in the list as purchased
- [X] Item should be shown as checked for 24 hours after the purchase is made (i.e. we assume the user does not need to buy the item again for at least 1 day

## Type of Changes

|     | Type                       |
| --- | -------------------------- |
|    | :bug: Bug fix              |
|  ✓ | :sparkles: New feature     |
|  ✓ | :hammer: Refactoring       |
|    | :100: Add tests            |
|  ✓  | :link: Update dependencies |
|    | :scroll: Docs              |

## Testing Steps / QA Criteria

* From the terminal, pull down this branch with `git pull origin `sd-vh-purchased-item` and check that branch out with `sd-vh-purchased-item`.
* Run `npm install` to install the newly added dependencies locally. Run `npm start` to launch the app.
* Navigate to the list by selecting a token from the available tokens or adding a new one.
* To mark an item as purchased, click on the checkbox next to it.
* Items purchased less than 24 hours ago would be highlighted and crossed.

## Before

Screen Shot 2020-08-30 at 10 03 08 AM

## After

Screen Shot 2020-08-30 at 9 59 54 AM

### CODE REVIEW

File: src/services/Listener/Listener.service.jsx
Comment (joeylaguna) [inline src/services/Listener/Listener.service.jsx:20]: Great approach!

**Review by joeylaguna — APPROVED** [review]
Looks great to me! Love the approach you used to check if 24 hours has passed. I'm going to approve it, and once your peers review, you can merge! Good job :D

File: src/components/AddItem/AddItem.component.jsx
Comment (CodeMeKathy) [inline src/components/AddItem/AddItem.component.jsx:4]: Thank you for updating. Consist naming convention and utilization of existing code base is important to remember.

File: src/components/Item/Item.component.jsx
Comment (CodeMeKathy) [inline src/components/Item/Item.component.jsx]: Have we addressed the matter of establishing naming conventions for `classNames` when styling? For example, using, `.Item__hightlight`?

File: src/services/Listener/Listener.service.jsx
Comment (CodeMeKathy) [reply src/services/Listener/Listener.service.jsx:20]: Ditto! Great job!

**Review by CodeMeKathy — APPROVED** [review]
A smart and logical approach to complete the issue acceptance criteria! Great Job!

File: src/components/Item/Item.component.jsx
Comment (daredtech) [reply src/components/Item/Item.component.jsx]: Thank you for pointing that out. I changed that.

**Review by brewswain — APPROVED** [review]
There really isn't much to say but excellent job! The date logic is really elegantly done, and the checkbox ui is really nice too!

**Review by jendevelops — APPROVED** [review]
Everything is looking good! The only warning (not a blocker) I see is that some component does not have a key prop. 
Could we take a look at eliminating that warning?

Screenshot from 2020-09-02 18-12-44

**Pull request merged**
